Library/Homebrew/cmd/style.rb

```diff
@@ -8,18 +19,38 @@ def style
       ARGV.formulae.map(&:path)
     end
 
+    Homebrew.failed = check_style(target, :fix => ARGV.flag?("--fix"))
+  end
+
+  # Checks style for a list of files
+  # If no block is given, just sends output to stdout
+  # If a block is given, passes an array of violations to the block.
+  # Returns true if violations were found, false otherwise.
+  def check_style(files, options = {})
+    files = [files] unless files.is_a? Array
+    fix = options[:fix]
     Homebrew.install_gem_setup_path! "rubocop", "0.39"
 
     args = [
       "--format", "simple", "--force-exclusion", "--config",
-      "#{HOMEBREW_LIBRARY}/.rubocop.yml",
+      "#{HOMEBREW_LIBRARY}/.rubocop.yml"
     ]
 
-    args << "--auto-correct" if ARGV.homebrew_developer? && ARGV.flag?("--fix")
+    args << "--auto-correct" if ARGV.homebrew_developer? && fix
 
-    args += target
+    args += files
 
-    system "rubocop", *args
-    Homebrew.failed = !$?.success?
+    if block_given?
+      txt = nil
+      Utils.popen(["rubocop", *args], "r") { |io| txt = io.read }
+      lines = txt.split("\n")
+      # Last two lines are summary, first line, if any, is header
+      lines[-2..-1] = []
+      lines.delete_at(0) unless lines.empty?
+      yield lines
```

Comment:
Rubocop supports JSON output. c.f. https://github.com/bbatsov/rubocop#formatters

I wonder should we use that instead of trying to parsing the output by ourselves.